lib/vsopdf/vso_appointment_form.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'pdf_forms'
+require 'tempfile'
+require 'securerandom'
+
+class VsoAppointmentForm
+  def initialize(appt)
+    @appt = appt
+  end
+
+  def to_pdf_args
+    {
+      "nameofvet": @appt.veteranFullName,
+      "SSNno": @appt.veteranSSN,
+      "filenumber": @appt.vaFileNumber,
+      "insno": @appt.insuranceNumber,
+      "claimantname": @appt.claimantFullName,
+      "address": @appt.claimantAddress,
+      "emailaddress": @appt.claimantEmail,
+      "daytime": @appt.claimantDaytimePhone,
+      "eveningphonenumber": @appt.claimantEveningPhone,
+      "relationship": @appt.relationship,
+      "Date@appt": @appt.appointmentDate,
+      "nameofservice": @appt.organizationName,
+      "e-mailaddressoftheorganizationnamedinitem3a": @appt.organizationEmail,
+      "jobtitile": "#{@appt.organizationRepresentativeName}, #{@appt.organizationRepresentativeTitle}",
+      "drugabuse": @appt.disclosureExceptionDrugAbuse ? 1 : 0,
+      "alcoholismoralcoholabuse": @appt.disclosureExceptionAlcoholism ? 1 : 0,
+      "infectionwiththehumanimmunodeficiencyvirushiv": @appt.disclosureExceptionHIV ? 1 : 0,
+      "sicklecellanemia": @appt.disclosureExceptionSickleCellAnemia ? 1 : 0
+    }.map { |k, v| ["F[0].Page_1[0].#{k}[0]", v] }.to_h
+  end
+
+  def generate_pdf
+    tmpf = Tempfile.new(['vsopdf', '.pdf'])
+    PdfForms.new(Settings.binaries.pdftk).fill_form 'lib/vsopdf/VBA-21-22-ARE.pdf', tmpf.path, to_pdf_args
+    tmpf.close
+    tmpf.path
+  end
+
+  def get_metadata(path)
+    {
+      "numberAttachments": 0,
+      "veteranFirstName": @appt.veteranFullName.split(' ').shift,
+      "veteranLastName": @appt.veteranFullName.split(' ').pop,
+      "source": 'Vets.gov',
+      "uuid": SecureRandom.uuid,
+      "zipCode": '10001', # TODO: get the actual zip code
+      "receiveDt": Time.zone.now.strftime('%Y-%m-%d %H:%M:%S'),
+      "fileNumber": @appt.vaFileNumber,
+      "hashV": Digest::SHA256.file(path).hexdigest,
+      "docType": 'burial',
+      "numberPages": 2
+    }
+  end
+
+  def send_pdf
+    path = generate_pdf
+    conn = Faraday.new("https://#{Settings.pension_burial.upload.host}") do |fd|
```

Comment:
This will be more clear once we have the development docs updated but we generally have a service class that inherits from [`Common::Client::Base`](https://github.com/department-of-veterans-affairs/vets-api/blob/063189e183ed02d7ebb000a31b60593bb4a20d7f/lib/common/client/base.rb#L13) e.g. as with the [`PensionBurial::Service`](https://github.com/department-of-veterans-affairs/vets-api/blob/1842af3f6f1fedc186349e056a8cc954b2c818b4/lib/pension_burial/service.rb#L4) 

The options for a service's Faraday connection are applied in a [configuration file](https://github.com/department-of-veterans-affairs/vets-api/blob/847c653e8bb65209e25a49222c99f4b575f8fa0e/lib/pension_burial/configuration.rb) and loaded via the service's [configuration method](https://github.com/department-of-veterans-affairs/vets-api/blob/1842af3f6f1fedc186349e056a8cc954b2c818b4/lib/pension_burial/service.rb#L5).

As this is a form submission it could be sent via a background that will retry until it succeeds, we use [sidekiq](https://github.com/mperham/sidekiq) for background processing and pension and burial has [an example](https://github.com/department-of-veterans-affairs/vets-api/blob/1842af3f6f1fedc186349e056a8cc954b2c818b4/app/workers/submit_saved_claim_job.rb#L11-L32) of calling a service from within a job.